carma_wm/test/MapConformerTest.cpp

```diff
@@ -175,5 +180,128 @@ TEST(MapConformer, ensureCompliance)
     ASSERT_EQ(count, ll.id());
     count++;
   }
+
+//Test Speed Limits
+/*
+  * Create lanelet map by hand
+  */
+ ROS_INFO_STREAM("Create new map");
+std::vector<lanelet::Lanelet> llts;
+lanelet::Areas areas;
+  // Linestring points
+  std::vector<lanelet::Point3d> pl, pm, pr;
+  pl.push_back (carma_wm::getPoint(1, 0 , 0));
+  pl.push_back (carma_wm::getPoint(1, 9, 0));
+  pl.push_back (carma_wm::getPoint(1, 18, 0));
+  pl.push_back (carma_wm::getPoint(10, 27, 0)); //45deg diag ll
+  pm.push_back (carma_wm::getPoint(9, 0, 0));
+  pm.push_back (carma_wm::getPoint(9, 9, 0));
+  pm.push_back (carma_wm::getPoint(9, 18, 0));
+  pm.push_back (carma_wm::getPoint(18, 27, 0)); //45deg diag ll
+  pr.push_back (carma_wm::getPoint(17, 0, 0));
+  pr.push_back (carma_wm::getPoint(17, 9 , 0));
+  pr.push_back (carma_wm::getPoint(17, 18, 0));
+  pr.push_back (carma_wm::getPoint(26, 27, 0)); //45deg diag ll
+
+  // Unique ids for line strings
+  std::vector<lanelet::Id> unique_ids;
+  for (int i = 0; i < 9; i ++)
+      unique_ids.push_back(lanelet::utils::getId());
+
+  // Create linestrings
+  lanelet::LineString3d left_1(unique_ids[0], { pl[0], pl[1] });
+  lanelet::LineString3d left_2(unique_ids[2], { pl[1], pl[2] });
+  lanelet::LineString3d left_3(unique_ids[6], { pl[2], pl[3] });
+  lanelet::LineString3d mid_1(unique_ids[1], { pm[0], pm[1]});
+  lanelet::LineString3d mid_2(unique_ids[3], { pm[1], pm[2]});
+  lanelet::LineString3d mid_3(unique_ids[7], { pm[2], pm[3]});
+  lanelet::LineString3d right_1(unique_ids[4], { pr[0], pr[1]});
+  lanelet::LineString3d right_2(unique_ids[5], { pr[1], pr[2]});
+  lanelet::LineString3d right_3(unique_ids[8], { pr[2], pr[3]});
+
+  // Create Lanelets for the maps
+
+  /*Map2*/
+  //First lanelet
+  llts.push_back(carma_wm::getLanelet(left_1, mid_1, lanelet::AttributeValueString::SolidSolid, lanelet::AttributeValueString::Dashed));
+
+  //Add regualtory element - Digital Speed Limit 1
+  std::shared_ptr<lanelet::DigitalSpeedLimit> dsl(new lanelet::DigitalSpeedLimit(lanelet::DigitalSpeedLimit::buildData(
+        lanelet::utils::getId(), 63_mph, llts, {}, {lanelet::Participants::Vehicle})));
+        llts.back().addRegulatoryElement(dsl);
+
+
+  //Second Lanelet
+  llts.push_back(carma_wm::getLanelet(mid_1, right_1, lanelet::AttributeValueString::Dashed, lanelet::AttributeValueString::SolidSolid));
+
+  // Add regualtory element - Digital Speed Limit 2
+      std::shared_ptr<lanelet::DigitalSpeedLimit> dsl2(new lanelet::DigitalSpeedLimit(lanelet::DigitalSpeedLimit::buildData(
+        lanelet::utils::getId(), 39_mph, llts, {}, {lanelet::Participants::Vehicle})));
+        llts.back().addRegulatoryElement(dsl2);
+
+  ASSERT_EQ(2, llts.size());//Assert that there are 2 lanelets
+  ROS_INFO_STREAM("There are "<<llts.size()<<" lanelets.");
+
+  ROS_ERROR_STREAM("SpeedLimitTestcheck");
```

Comment:
please remove any ERROR statements from the test as it will be confusing to the following developer